File: src/main/java/Ex04_Equals.java
import java.util.List;

import static java.util.Collections.singletonList;

public class Ex04_Equals {
    public static void main(String[] args) {
        int refInt1 = 0;
        int refInt2 = 0;
        System.out.println(refInt1 == refInt2);

        List<String> ref1 = singletonList("Hello");
        List<String> ref2 = singletonList("Hello");
        System.out.println(ref1 == ref2); // To check references
        System.out.println(ref1.equals(ref2)); // To check equality
    }
}

File: src/main/kotlin/Ex04_Equals.kt
fun main() {
  val refInt1 = 0
  val refInt2 = 0
  println(refInt1 == refInt2) // To check equality

  val ref1: List<String> = listOf("Hello")
  val ref2: List<String> = listOf("Hello")

  println(ref1 === ref2) // To check references
  println(ref1 == ref2) // To check equality
}